### ISSUE TITLE
SystemError: Objects/moduleobject.c:50: bad argument to internal function

### ISSUE THREAD

**Issue by OkanEsen** [snare/voltron#128]
After I run the target voltron stops working and gives me this error:

```
Voltron loaded.
(lldb) target create "malloctest"
Current executable set to 'malloctest' (x86_64).
(lldb) run
SystemError: Objects/moduleobject.c:50: bad argument to internal function
error: Aborting reading of commands after command #0: 'voltron stopped' failed with error: unable to execute script function

```

debugger.log

```
2016-04-12 14:48:32,648 INFO     __init__.py:84          setup_logging -- ======= VOLTRON - DEFENDER OF THE UNIVERSE [debug log] =======
2016-04-12 14:48:32,648 DEBUG      plugin.py:83        register_plugin -- Ignoring invalid plugin: <class 'voltron.plugin.VoltronPlugin'>
2016-04-12 14:48:32,648 DEBUG      plugin.py:83        register_plugin -- Ignoring invalid plugin: <class 'voltron.plugin.APIPlugin'>
2016-04-12 14:48:32,648 DEBUG      plugin.py:83        register_plugin -- Ignoring invalid plugin: <class 'voltron.plugin.DebuggerAdaptorPlugin'>
2016-04-12 14:48:32,648 DEBUG      plugin.py:83        register_plugin -- Ignoring invalid plugin: <class 'voltron.plugin.ViewPlugin'>
2016-04-12 14:48:32,648 DEBUG      plugin.py:83        register_plugin -- Ignoring invalid plugin: <class 'voltron.plugin.WebPlugin'>
2016-04-12 14:48:32,648 DEBUG      plugin.py:83        register_plugin -- Ignoring invalid plugin: <class 'voltron.plugin.CommandPlugin'>
2016-04-12 14:48:32,648 DEBUG      plugin.py:66        register_plugin -- Registering API plugin: <class 'breakpoints.APIBreakpointsPlugin'>
2016-04-12 14:48:32,648 DEBUG      plugin.py:66        register_plugin -- Registering API plugin: <class 'command.APICommandPlugin'>
2016-04-12 14:48:32,648 DEBUG      plugin.py:66        register_plugin -- Registering API plugin: <class 'dereference.APIDerefPlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:66        register_plugin -- Registering API plugin: <class 'disassemble.APIDisassemblePlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:66        register_plugin -- Registering API plugin: <class 'memory.APIReadMemoryPlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:66        register_plugin -- Registering API plugin: <class 'plugins.APIPluginsPlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:66        register_plugin -- Registering API plugin: <class 'registers.APIRegistersPlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:66        register_plugin -- Registering API plugin: <class 'stack.APIStackPlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:66        register_plugin -- Registering API plugin: <class 'state.APIStatePlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:66        register_plugin -- Registering API plugin: <class 'targets.APITargetsPlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:66        register_plugin -- Registering API plugin: <class 'version.APIVersionPlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:69        register_plugin -- Registering debugger plugin: <class 'dbg_lldb.LLDBAdaptorPlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:69        register_plugin -- Registering debugger plugin: <class 'dbg_mock.MockAdaptorPlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:72        register_plugin -- Registering view plugin: <class 'backtrace.BacktraceViewPlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:72        register_plugin -- Registering view plugin: <class 'breakpoints.BreakpointsViewPlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:72        register_plugin -- Registering view plugin: <class 'command.CommandViewPlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:72        register_plugin -- Registering view plugin: <class 'disasm.DisasmViewPlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:72        register_plugin -- Registering view plugin: <class 'memory.MemoryViewPlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:72        register_plugin -- Registering view plugin: <class 'memory.StackViewPlugin'>
2016-04-12 14:48:32,649 DEBUG      plugin.py:72        register_plugin -- Registering view plugin: <class 'register.RegisterViewPlugin'>
2016-04-12 14:48:32,649 DEBUG    dbg_lldb.py:35               __init__ -- lldb.debugger is valid - probably running inside LLDB
2016-04-12 14:48:32,658 DEBUG        core.py:133          run_listener -- Starting listener for tcp socket on ['127.0.0.1', 5555, <werkzeug.wsgi.DispatcherMiddleware object at 0x112b70710>]
2016-04-12 14:48:32,661 DEBUG        core.py:133          run_listener -- Starting listener for domain socket on ['/Users/Okan/.voltron/sock', <werkzeug.wsgi.DispatcherMiddleware object at 0x112b70710>]
2016-04-12 14:48:32,721 DEBUG        core.py:185        handle_request -- data = {"request": "disassemble", "type": "request", "timeout": 10, "block": true, "data": {"count": 11, "address": null, "target_id": 0, "use_capstone": false}}
2016-04-12 14:48:32,739 DEBUG        core.py:185        handle_request -- data = {"request": "memory", "type": "request", "timeout": 10, "block": true, "data": {"register": "sp", "target_id": 0, "length": null, "command": null, "words": 7, "address": null, "deref": true}}
2016-04-12 14:48:32,739 DEBUG        core.py:185        handle_request -- data = {"request": "targets", "type": "request", "timeout": 10, "block": true, "data": {}}
2016-04-12 14:48:32,747 DEBUG        core.py:185        handle_request -- data = {"request": "targets", "type": "request", "timeout": 10, "block": true, "data": {}}
2016-04-12 14:48:32,748 DEBUG        core.py:185        handle_request -- data = {"request": "registers", "type": "request", "timeout": 10, "block": true, "data": {"thread_id": null, "registers": [], "target_id": 0}}
2016-04-12 14:48:32,749 DEBUG        core.py:185        handle_request -- data = {"request": "disassemble", "type": "request", "timeout": 10, "block": true, "data": {"count": 1, "address": null, "target_id": 0, "use_capstone": null}}
2016-04-12 14:48:32,822 DEBUG        core.py:185        handle_request -- data = {"request": "command", "type": "request", "timeout": 10, "block": true, "data": {"command": "bt"}}
2016-04-12 14:48:32,823 DEBUG        core.py:185        handle_request -- data = {"request": "registers", "type": "request", "timeout": 10, "block": false, "data": {"thread_id": null, "registers": ["pc"], "target_id": 0}}
2016-04-12 14:48:32,823 DEBUG        core.py:247      dispatch_request -- Dispatching request: {"request": "registers", "type": "request", "timeout": 10, "block": false, "data": {"thread_id": null, "registers": ["pc"], "target_id": 0}}
2016-04-12 14:48:32,824 DEBUG        core.py:265      dispatch_request -- Response: {"status": "error", "data": {"message": "No such target", "code": 4101}, "type": "response"}
2016-04-12 14:48:32,824 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:32] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:32,830 DEBUG        core.py:185        handle_request -- data = {"request": "breakpoints", "type": "request", "timeout": 10, "block": true, "data": {}}
2016-04-12 14:48:42,727 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:42] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:42,741 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:42] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:42,742 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:42] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:42,745 DEBUG        core.py:185        handle_request -- data = {"request": "disassemble", "type": "request", "timeout": 10, "block": true, "data": {"count": 11, "address": null, "target_id": 0, "use_capstone": false}}
2016-04-12 14:48:42,748 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:42] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:42,749 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:42] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:42,750 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:42] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:42,757 DEBUG        core.py:185        handle_request -- data = {"request": "memory", "type": "request", "timeout": 10, "block": true, "data": {"register": "sp", "target_id": 0, "length": null, "command": null, "words": 7, "address": null, "deref": true}}
2016-04-12 14:48:42,758 DEBUG        core.py:185        handle_request -- data = {"request": "targets", "type": "request", "timeout": 10, "block": true, "data": {}}
2016-04-12 14:48:42,762 DEBUG        core.py:185        handle_request -- data = {"request": "disassemble", "type": "request", "timeout": 10, "block": true, "data": {"count": 1, "address": null, "target_id": 0, "use_capstone": null}}
2016-04-12 14:48:42,763 DEBUG        core.py:185        handle_request -- data = {"request": "registers", "type": "request", "timeout": 10, "block": true, "data": {"thread_id": null, "registers": [], "target_id": 0}}
2016-04-12 14:48:42,763 DEBUG        core.py:185        handle_request -- data = {"request": "targets", "type": "request", "timeout": 10, "block": true, "data": {}}
2016-04-12 14:48:42,827 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:42] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:42,831 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:42] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:42,838 DEBUG        core.py:185        handle_request -- data = {"request": "registers", "type": "request", "timeout": 10, "block": false, "data": {"thread_id": null, "registers": ["pc"], "target_id": 0}}
2016-04-12 14:48:42,838 DEBUG        core.py:247      dispatch_request -- Dispatching request: {"request": "registers", "type": "request", "timeout": 10, "block": false, "data": {"thread_id": null, "registers": ["pc"], "target_id": 0}}
2016-04-12 14:48:42,839 DEBUG        core.py:265      dispatch_request -- Response: {"status": "success", "data": {"registers": {}}, "type": "response"}
2016-04-12 14:48:42,839 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:42] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:42,841 DEBUG        core.py:185        handle_request -- data = {"request": "command", "type": "request", "timeout": 10, "block": true, "data": {"command": "bt"}}
2016-04-12 14:48:42,844 DEBUG        core.py:185        handle_request -- data = {"request": "breakpoints", "type": "request", "timeout": 10, "block": true, "data": {}}
2016-04-12 14:48:52,746 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:52] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:52,758 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:52] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:52,763 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:52] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:52,763 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:52] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:52,764 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:52] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:52,764 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:52] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:52,766 DEBUG        core.py:185        handle_request -- data = {"request": "disassemble", "type": "request", "timeout": 10, "block": true, "data": {"count": 11, "address": null, "target_id": 0, "use_capstone": false}}
2016-04-12 14:48:52,777 DEBUG        core.py:185        handle_request -- data = {"request": "targets", "type": "request", "timeout": 10, "block": true, "data": {}}
2016-04-12 14:48:52,779 DEBUG        core.py:185        handle_request -- data = {"request": "targets", "type": "request", "timeout": 10, "block": true, "data": {}}
2016-04-12 14:48:52,779 DEBUG        core.py:185        handle_request -- data = {"request": "registers", "type": "request", "timeout": 10, "block": true, "data": {"thread_id": null, "registers": [], "target_id": 0}}
2016-04-12 14:48:52,780 DEBUG        core.py:185        handle_request -- data = {"request": "memory", "type": "request", "timeout": 10, "block": true, "data": {"register": "sp", "target_id": 0, "length": null, "command": null, "words": 7, "address": null, "deref": true}}
2016-04-12 14:48:52,781 DEBUG        core.py:185        handle_request -- data = {"request": "disassemble", "type": "request", "timeout": 10, "block": true, "data": {"count": 1, "address": null, "target_id": 0, "use_capstone": null}}
2016-04-12 14:48:52,844 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:52] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:52,845 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:52] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:52,850 DEBUG        core.py:185        handle_request -- data = {"request": "registers", "type": "request", "timeout": 10, "block": false, "data": {"thread_id": null, "registers": ["pc"], "target_id": 0}}
2016-04-12 14:48:52,850 DEBUG        core.py:247      dispatch_request -- Dispatching request: {"request": "registers", "type": "request", "timeout": 10, "block": false, "data": {"thread_id": null, "registers": ["pc"], "target_id": 0}}
2016-04-12 14:48:52,850 DEBUG        core.py:265      dispatch_request -- Response: {"status": "success", "data": {"registers": {}}, "type": "response"}
2016-04-12 14:48:52,851 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:48:52] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:48:52,856 DEBUG        core.py:185        handle_request -- data = {"request": "command", "type": "request", "timeout": 10, "block": true, "data": {"command": "bt"}}
2016-04-12 14:48:52,857 DEBUG        core.py:185        handle_request -- data = {"request": "breakpoints", "type": "request", "timeout": 10, "block": true, "data": {}}
2016-04-12 14:49:02,771 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:49:02] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:49:02,779 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:49:02] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:49:02,782 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:49:02] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:49:02,782 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:49:02] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:49:02,783 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:49:02] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:49:02,783 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:49:02] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:49:02,789 DEBUG        core.py:185        handle_request -- data = {"request": "disassemble", "type": "request", "timeout": 10, "block": true, "data": {"count": 11, "address": null, "target_id": 0, "use_capstone": false}}
2016-04-12 14:49:02,796 DEBUG        core.py:185        handle_request -- data = {"request": "disassemble", "type": "request", "timeout": 10, "block": true, "data": {"count": 1, "address": null, "target_id": 0, "use_capstone": null}}
2016-04-12 14:49:02,798 DEBUG        core.py:185        handle_request -- data = {"request": "targets", "type": "request", "timeout": 10, "block": true, "data": {}}
2016-04-12 14:49:02,799 DEBUG        core.py:185        handle_request -- data = {"request": "registers", "type": "request", "timeout": 10, "block": true, "data": {"thread_id": null, "registers": [], "target_id": 0}}
2016-04-12 14:49:02,800 DEBUG        core.py:185        handle_request -- data = {"request": "targets", "type": "request", "timeout": 10, "block": true, "data": {}}
2016-04-12 14:49:02,800 DEBUG        core.py:185        handle_request -- data = {"request": "memory", "type": "request", "timeout": 10, "block": true, "data": {"register": "sp", "target_id": 0, "length": null, "command": null, "words": 7, "address": null, "deref": true}}
2016-04-12 14:49:02,858 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:49:02] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:49:02,858 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:49:02] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:49:02,866 DEBUG        core.py:185        handle_request -- data = {"request": "registers", "type": "request", "timeout": 10, "block": false, "data": {"thread_id": null, "registers": ["pc"], "target_id": 0}}
2016-04-12 14:49:02,866 DEBUG        core.py:247      dispatch_request -- Dispatching request: {"request": "registers", "type": "request", "timeout": 10, "block": false, "data": {"thread_id": null, "registers": ["pc"], "target_id": 0}}
2016-04-12 14:49:02,866 DEBUG        core.py:265      dispatch_request -- Response: {"status": "success", "data": {"registers": {}}, "type": "response"}
2016-04-12 14:49:02,866 INFO    _internal.py:87                   _log -- 127.0.0.1 - - [12/Apr/2016 14:49:02] "POST /api/request HTTP/1.1" 200 -
2016-04-12 14:49:02,872 DEBUG        core.py:185        handle_request -- data = {"request": "command", "type": "request", "timeout": 10, "block": true, "data": {"command": "bt"}}
2016-04-12 14:49:02,873 DEBUG        core.py:185        handle_request -- data = {"request": "breakpoints", "type": "request", "timeout": 10, "block": true, "data": {}}

```

**Comment by snare** [comment]
Directly after that failure occurs, can you drop into the Python REPL within LLDB and do this:

    (lldb) script
    Python Interactive Interpreter. To exit, type 'quit()', 'exit()' or Ctrl-D.
    >>> import voltron
    >>> voltron.command.handle_command('stopped')
    >>>

Hopefully it'll log the whole backtrace instead of just the error like you're seeing.

**Comment by OkanEsen** [comment]
Alright, this is what I'm getting:

```
(lldb) script
Python Interactive Interpreter. To exit, type 'quit()', 'exit()' or Ctrl-D.
>>> import voltron
>>> voltron.command.handle_command('stopped')
>>>
(lldb) run
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Applications/Xcode.app/Contents/SharedFrameworks/LLDB.framework/Resources/Python/lldb/embedded_interpreter.py", line 105, in run_one_line
    repl.runsource (input_string)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/code.py", line 76, in runsource
    code = self.compile(source, filename, symbol)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/codeop.py", line 168, in __call__
    return _maybe_compile(self.compiler, source, filename, symbol)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/codeop.py", line 82, in _maybe_compile
    code = compiler(source, filename, symbol)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/codeop.py", line 135, in __call__
    if codeob.co_flags & feature.compiler_flag:
KeyboardInterrupt
SystemError: Objects/moduleobject.c:50: bad argument to internal function
error: Aborting reading of commands after command #0: 'voltron stopped' failed with error: unable to execute script function

```

**Comment by snare** [comment]
Sorry, if you could execute the commands in the REPL *after* the failure. Are you still able to use the LLDB prompt after the error?

**Comment by OkanEsen** [comment]
That's exactly what I did or I'm getting something wrong, sorry.

Here is the output (I removed the unnecessary parts):

```
❯ lldb malloctest
Voltron loaded.
(lldb) target create "malloctest"
Current executable set to 'malloctest' (x86_64).
(lldb) run
SystemError: Objects/moduleobject.c:50: bad argument to internal function
error: Aborting reading of commands after command #0: 'voltron stopped' failed with error: unable to execute script function

[...]

Process 4452 exited with status = 0 (0x00000000)
(lldb) script
Python Interactive Interpreter. To exit, type 'quit()', 'exit()' or Ctrl-D.
>>> import voltron
>>> voltron.command.handle_command('stopped')
>>>
(lldb) run
SystemError: Objects/moduleobject.c:50: bad argument to internal function
error: Aborting reading of commands after command #0: 'voltron stopped' failed with error: unable to execute script function

[...]

Process 4486 exited with status = 0 (0x00000000)

```

**Comment by snare** [comment]
Ah ok. Sorry, your previous message was confusing because the script commands appeared in the log before the `run` command.

Can you try debugging the test binary inside the Voltron package?

Kill all your views and just start a registers view:

    $ voltron v r

Then start the debugger with the test inferior:

    $ cc -o tests/inferior tests/inferior.c
    $ lldb tests/inferior
    (lldb) voltron init
    (lldb) b main
    (lldb) run

If the same error occurs, can you tell me what version of LLDB, Python, etc you're running? If you installed from PyPI can you try installing the version on GitHub and tell me if the error still occurs?

Thanks

**Comment by OkanEsen** [comment]
No worries!

There seems to be a strange issue, when I run the view segment for the registers (or any other). It just displays "No such target".

Running and debugging the test inferior gives me following output:

```
❯ lldb tests/inferior
Voltron loaded.
(lldb) target create "tests/inferior"
Current executable set to 'tests/inferior' (x86_64).
(lldb) voltron init
SystemError: Objects/moduleobject.c:50: bad argument to internal function
error: unable to execute script function
(lldb) b main
Breakpoint 1: where = inferior`main, address = 0x0000000100000cf0
(lldb) run
SystemError: Objects/moduleobject.c:50: bad argument to internal function
error: Aborting reading of commands after command #0: 'voltron stopped' failed with error: unable to execute script function
Process 11273 launched: '/Users/Okan/Desktop/voltron/tests/inferior' (x86_64)
SystemError: Objects/moduleobject.c:50: bad argument to internal function
error: Aborting reading of commands after command #0: 'voltron stopped' failed with error: unable to execute script function
Process 11273 stopped
* thread #1: tid = 0xf766b, 0x0000000100000cf0 inferior`main, queue = 'com.apple.main-thread', stop reason = breakpoint 1.1
    frame #0: 0x0000000100000cf0 inferior`main
inferior`main:
->  0x100000cf0 <+0>: pushq  %rbp
    0x100000cf1 <+1>: movq   %rsp, %rbp
    0x100000cf4 <+4>: subq   $0x50, %rsp
    0x100000cf8 <+8>: movl   $0x0, -0x4(%rbp)
(lldb)

```

> If the same error occurs, can you tell me what version of LLDB, Python, etc you're running?
LLDB: lldb-340.4.119
Python: 2.7.11 (installed with brew)
Mac: 10.11.4

> If you installed from PyPI can you try installing the version on GitHub and tell me if the error still occurs?

Tried both, with pip and with the version on GitHub, same error unfortunately.

**Comment by snare** [comment]
It will display "no target" until a debug target is available in the debugger, but if the "voltron stopped" command isn't working at all, then the views will never update.

This bit here is pretty crazy, and seems like a simpler way to reproduce:

```
(lldb) voltron init
SystemError: Objects/moduleobject.c:50: bad argument to internal function
```

If you just launch the debugger and do this what happens:

```
(lldb) script
Python Interactive Interpreter. To exit, type 'quit()', 'exit()' or Ctrl-D.
>>> import voltron
>>> voltron.command.handle_command('init')
Registered stop-hook
>>> ^D
(lldb) voltron init
Registered stop-hook
```

**Comment by OkanEsen** [comment]
Same error:

```
Voltron loaded.
(lldb) script
Python Interactive Interpreter. To exit, type 'quit()', 'exit()' or Ctrl-D.
>>> import voltron
>>> voltron.command.handle_command('init')
Registered stop-hook
>>>
(lldb) voltron init
SystemError: Objects/moduleobject.c:50: bad argument to internal function
error: unable to execute script function
(lldb)

```

**Comment by snare** [comment]
That's bizarre. `voltron.command.handle_command('init')` is pretty much the entire code path of `voltron init`. It seems that LLDB is failing to even execute the `voltron` command for some reason. I'm currently using a different LLDB version, but I'll see if I can reproduce it here on the same version.

**Comment by snare** [comment]
Seems to be specific to that version of LLDB. Works fine with the version included in the Swift 2.2 distribution, which is what I was using, but I can reproduce it with `lldb-340.4.119`. I'll investigate further.

**Comment by snare** [comment]
Ok, should be fixed in 2f6cad7faae7ed78450eee8941382de4bf176481.

**Comment by OkanEsen** [comment]
Awesome, works fine now, thanks! :+1:
